Reports documentation update (#4143)
chef/automate: components/automate-chef-io/content/docs/reports.md

    weight = 10
+++

> The _Reports_ page under the _Compliance_ dashboard provides comprehensive insight into the compliance status of all scanned infrastructure.

Reviewer: ```suggestion
The _Reports_ page under the _Compliance_ ( Compliance > Reports) dashboard provides comprehensive insight into the compliance status of all scanned infrastructure.
```
What do you think of adding this sort of breadcrumb in the docs?

Author: I love this idea!

The dashboard shows the results of all scans with end times on the _currently selected day_.
The currently selected day, as highlighted in the search bar, bases itself on timestamps in Coordinated Universal Time (UTC).

The trend graph provides a historical view, which can change to show a 10 day, 1 month, 3 month, or 1 year overview.

To view scan reports in the past, users can select a different date from the calendar located in the search bar.

![Compliance Date Selector](/images/docs/compliance-date.png)

### Compliance Data Sources

> Users with audit cookbook configurations can expect to see the results of those scans land in this view.

Author: Remember how I mentioned to you that there are sections in this page, not related to the UI, that are just plopped in here for no apparent reason? This is one of them.

> Users with audit cookbook configurations can expect to see the results of those scans land in this view.

Reviewer: Rip them out and rehome them.

For example, you can search for `Node Name: prod*` and filter the results to list all the node names that begin with `prod`.

Profile
: Filter your results by profile. This list will include any executed profiles, which are profiles with received scan results.

Node

> : Filter your results by node item. A node is any scannable for which we have received scan results, which can be a Chef node, an aws node, an aws account region, or any other target Chef InSpec supports.

Reviewer: `any scannable for which`

### Deep Filtering

> Deep Filtering allows you to view the state of your infrastructure from the perspective of a single profile, or a single profile and one of its child controls.

Reviewer: Both, I believe, but revise it if you don't think it catches the nuances.

### Waivers

A node's waived status appears if applicable in displays where a node's status appears in Chef Automate.
The Compliance Reports overview displays the node count and history of waived nodes, and the count and history of waived controls.

> _Nodes_ and _Profiles_ views include _Waived Nodes_ and _Waived Profiles_ status filters respectively.

Author: This section is done as a stop-gap documentation measure because waivers were originally added without documentation. I would greatly like to revise and add to this further.

> _Nodes_ and _Profiles_ views include _Waived Nodes_ and _Waived Profiles_ status filters respectively.

Reviewer: It's possible that the product has changed enough that we need to review the need for this page to exist.

### Compliance Reports Results and Job ID Filters

> Chef Automate users can see the results of their scan jobs from https://{{< example_fqdn "automate" >}}/compliance/scan-jobs/jobs by selecting _Report_, which will redirect to the compliance tab.

Author: I believe those are the directions. Honestly, this section should just be moved to Scan Jobs, especially given the contents of the screenshot below.

> Chef Automate users can see the results of their scan jobs from https://{{< example_fqdn "automate" >}}/compliance/scan-jobs/jobs by selecting _Report_, which will redirect to the compliance tab.

Reviewer: OK.

This Reports view uses the _job id_ filter as well as any additional filters, and ignores the _end time_ filter.

> ![Getting to Compliance from a Scan Job](/images/docs/compliance-jobid.png)

Author: Oh, it is most definitely not a current image. Will be fixed.

> ![Getting to Compliance from a Scan Job](/images/docs/compliance-jobid.png)

Author: Well, this is awkward. This image is now the only image not updated.

## Compliance Status and Report Metadata

The _Compliance Status and Report Metadata_ bar resides directly beneath the search bar.

> Selecting anywhere on the compliance status bar expands the `Report Metadata` information, which summarizes the nodes, report date, duration, status, number of platforms, number of environments, and number of profiles used in your scan.

Author: `Report Metadata` is capitalized and listed as such in the UI.

![Compliance Metadata](/images/docs/compliance-metadata.png)

## Compliance Overview

> The Compliance Overview provides insights into the status of your system.

Author: Can I just delete the sentence?

> The Compliance Overview provides insights into the status of your system.

Reviewer: Sure!

Toggle between _Node Status_ and _Profile Status_ to view your system's overall compliance.

### Node Status

> The _Node Status_ view provides insight into your system's compliance status from the operational perspective of nodes.

Author: how

> The _Node Status_ view provides insight into your system's compliance status from the operational perspective of nodes.

Author: P.S. Given the number of "passive" notes on this, this sentence structure is clearly used formulaically through this page.

> The _Node Status_ view provides insight into your system's compliance status from the operational perspective of nodes.

Reviewer: And it is still passive (do you see how yet?).

Node Status
: Visualizes the quantity of nodes that have passed and failed compliance scans, as well as the number of skipped or waived nodes.

Severity of Node Failures
: Indicates the severity of the compliance scan failures on your system's nodes.

Node Status Over Time
: Represents the changes in size and compliance status of your system over times. The dropdown menu in the upper left corner of the visualization allows you to change the date range represented in this view. Modifying the date range will affect the display of the other visualizations.

Top Platform Failures

> : Represents the amount of compliance scan failures ranked by operating system. Hovering over an individual bubble shows the platform name and the number of impacted nodes.

Author: Well, we have a bit of a conundrum here. 
The UI labels this interface as "Top Platform Failures."
If you click on the bubble, such as `ubuntu`, it applies a filter of "platform_with_version:ubuntu" and shows a list of relevant nodes.
I tried to have at least one mention of "operating system" in the first sentence, but everything else in the UI screams "Platform."

> : Represents the amount of compliance scan failures ranked by operating system. Hovering over an individual bubble shows the platform name and the number of impacted nodes.

Author: See also: https://github.com/chef/automate/pull/4143/files#r468149509
for an image of why "bubble" is continually used.

 Top Environment Failures

> : Represents the amount of compliance scan failures ranked by environment. Hovering over an individual bubble shows the environment name and the number of impacted nodes.

Reviewer: I don't know why we used amount (mists of time, again) but this is the wrong context. It pertains more to volume than count. "A pinch of salt" or "a cup of sugar" are amounts.

### Profile Status

> The _Profile Status_ view provides insight into your system's compliance status from the compliance perspective of Compliance profiles run during scans.

Reviewer: passive.

Control Status

> : Displays the number of controls in your system, organized by passing, failing, and skipped controls.

Reviewer: ```suggestion
: shows the number of controls in your system, grouped by passing, failing, and skipped controls.
```

Severity of Control Failures

> : Displays a ranked view of the number and significance of failed controls, sorted into critical, major, and minor control failures.

Reviewer: ```suggestion
: Shows failed controls, ranked by number and importance of the control failure, grouped as critical, major, and minor.
```

Control Status Over Time

> : A line graph representing the quantity of controls and Compliance scan results over time. Modify the time-frame using the calendar button in the upper right corner of the chart.

Reviewer: ```suggestion
: A line graph showing the the number of controls and Compliance scan results over time. Use the calendar button in the upper right corner of the chart to change the time frame.
```

Top Profile Failures

> : Indicates the profiles with the highest failure rate. Hovering over an individual bubble shows the full name of the profile and the number of impacted nodes.

Reviewer: ```suggestion
: Indicates the profiles with the highest failure rate. Hover over an individual bubble to show the full profile name and the number of impacted nodes.
```
I like indicates here because the relationship is _indexical_, meaning it points to other things. Are impacted nodes failed nodes, or node with the profile?

Author: It refers to failed nodes.
When "profile" is mentioned here, it refers to the operating system, but the graph is labeled "Top Profile Failures."
<img width="299" alt="Chef_Automate" src="https://user-images.githubusercontent.com/5807053/89825578-ce9d2e80-db22-11ea-912d-53f4730b7c16.png">

Top Control Failures

> : The Top Control Failures chart illustrates the most frequently failing controls. Hovering over an individual bubble shows the name of the control and the number of impacted nodes.

Reviewer: ```suggestion
: Indicates the most frequently failing controls. Hover over an individual bubble to show the control name and the number of impacted nodes.
```

## Switching Views

> You may also switch your views to see compliance report results from the perspective of _Nodes_, _Profiles_, and _Controls_.

Reviewer: How do I do this amazing switching thing?

### Nodes

> The _Nodes_ view provides more detailed insight into the compliance status of the nodes in your system.

Reviewer: This section was bad when I (presumably) wrote it and it is still bad now. Revise to...anything is better.

Author: ![tenor-179266690](https://user-images.githubusercontent.com/5807053/89831187-643cbc00-db2b-11ea-862a-df85b863543f.gif)

: Time in hours, days, or months since the last scan on that node.

Control Failures

> : Shows the number of failing controls if any.

Reviewer: ```suggestion
: Shows the number of failing controls, if any.
```

Filter

> : Select a node by selecting the filter icon on the right side of the row.

Reviewer: ```suggestion
: Select the filter icon on the right side of the row to select a node.
```

More Options

> : Select the More Options icon (...) to display a menu with two choices: _Add Filter_ and _Scan Results_. _Add Filter_ applies an `id` filter of the selected node to the results. _Scan Results_ opens a side-window and displays the scan results of controls for the specific node.

Reviewer: ```suggestion
: Select the (...) icon to display a menu with two additional options: _Add Filter_ and _Scan Results_. Use _Add Filter_ to narrow the results to the `id` of the selected node. Use _Scan Results_ to show the controls used in the last compliance scan of the selected node in a side window.
```
